### PR TITLE
support tcc branch commit & rollback

### PR DESCRIPTION
**Pull request by bohehe**
ref: https://github.com/cectc/dbpack/issues/12

### Ⅰ. Describe what this PR did
support tcc mode distributed transaction

### Ⅱ. Does this pull request fix one issue?
#12

### CODE REVIEW

File: pkg/dt/distributed_transaction_manger.go [hunk 1]
```diff
@@ -322,8 +327,16 @@ func (manager *DistributedTransactionManager) processNextBranchSession(ctx conte
...
+		status api.BranchSession_BranchStatus
+		err    error
+	)
 	if bs.Status == api.PhaseTwoCommitting {
-		status, err := manager.branchCommit(bs)
+		if bs.Type == api.TCC {
+			status, err = manager.tccBranchCommit(bs)
+		} else {
+			status, err = manager.branchCommit(bs)
+		}
```
Comment (dk-lockdown) [inline pkg/dt/distributed_transaction_manger.go]: this if judgment logic should be placed in the method branchCommit

File: pkg/dt/distributed_transaction_manger.go [hunk 2]
```diff
@@ -340,7 +353,11 @@ func (manager *DistributedTransactionManager) processNextBranchSession(ctx conte
 				}
 			}
 		} else {
-			status, err := manager.branchRollback(bs)
+			if bs.Type == api.TCC {
+				status, err = manager.tccBranchRollback(bs)
+			} else {
+				status, err = manager.branchRollback(bs)
+			}
```
Comment (dk-lockdown) [inline pkg/dt/distributed_transaction_manger.go]: this if judgment logic should be placed in the method branchRollback

File: pkg/dt/distributed_transaction_manger.go (hunk 1 above)
Comment (dk-lockdown) [reply pkg/dt/distributed_transaction_manger.go]: keep control loop simple

**Comment by bohehe** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pkg/dt/distributed_transaction_manger.go (hunk 2 above)
Comment (bohehe) [reply pkg/dt/distributed_transaction_manger.go]: Thanks, just adjusted.

**Review by dk-lockdown — APPROVED** [review]
lgtm

**Comment by fatelei** [comment]
LGTM

**Comment by allincache** [comment]
lgtm

**Comment by zhu733756** [comment]
lgtm

**Pull request merged**